packages/widget-chat/src/reducers/conversation.js

```diff
@@ -30,7 +38,25 @@ export default function conversation(state = {
     });
   }
 
-  default:
+  case RECEIVE_MERCURY_ACTIVITY: {
```

Comment:
I feel like it belongs in the reducer since the reducer is where we know what the current conversation is. that said, i haven't looked too closely at the overall architecture in a week or two. If the reducer sees that the message is for a different conversation, it can return the original state object unchanged (without copying it) and there be anything to trigger a rerender. (Rerenders are based on shallow comparisons - that's why we typically copy the current state onto a new object and also why I mention immutable from time to time)